### PR TITLE
NMS-14084: Add the ability to define an enumeration to convert collected strings into numeric values

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-14084

### CODE REVIEW

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+= Value Mapping Property extender
+The Value Mapping property extender provides a mechanism that works like a lookup table for values of a local MIB table column.
+
+== When to use the Value Mapping Property extender
+Use the Value Mapping property extender to map an enumerated set of values (e.g. strings) to a corresponding set of values.
+This allows to persist changes of string enumeration values as integers over time as time-series.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:6]:
```suggestion
This allows OpenNMS to persist changes of string enumeration values as integers over time as time-series.
```
Object was missing. Not sure if OpenNMS is the right word here.

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+== Configure the Value Mapping Property Extender
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:8]:
```suggestion
== Configure the Value Mapping property extender
```

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+| Default value
+
+3+| *Required*
+
+| source-attribute
+| The alias name of the property used for the value lookup.
+| n/a
+
+| target-type
+| The target data-type, e.g. gauge.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:26]:
```suggestion
| The target data-type; for example, gauge.
```
This is where I think the "for example" rule is annoying. :-)

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 4]
```diff
@@ -149,3 +149,47 @@ With the configuration in place, you can test it using the `collect` command ava
 ----
 opennms:collect -n 1 org.opennms.protocols.xml.collector.XmlCollector 127.0.0.1 collection=xml-opennms-nodes
 ----
+
+== Mapping values
+
+Sometimes data is represented as string values.
+These values are normally not persisted as time-series data, this means changes are not visible over time.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:156]:
```suggestion
These values are normally not persisted as time-series data; this means changes are not visible over time.
```

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 5]
```diff
@@ -149,3 +149,47 @@ With the configuration in place, you can test it using the `collect` command ava
...
+To circumvent this, we allow to define mappings between input values and values to be persisted.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:157]:
```suggestion
To circumvent this, we allow mappings defined between input values and values to be persisted.
```

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 6]
```diff
@@ -149,3 +149,47 @@ With the configuration in place, you can test it using the `collect` command ava
...
+        <read>456</read>
+    </record>
+    <record>
+        <input>ccc</input>
+        <read>789</read>
+    </record>
+</records>
+----
+
+The following group configuration allows to persist the `input` values as integer values over time:
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:179]:
```suggestion
The following group configuration allows you to persist the `input` values as integer values over time:
```

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 7]
```diff
@@ -149,3 +149,47 @@ With the configuration in place, you can test it using the `collect` command ava
...
+        <xml-mapping from="aaa" to="10" /> <2>
+        <xml-mapping from="bbb" to="20" />
+        <xml-mapping to="1000" /> <3>
+    </xml-object>
+    <xml-object name="read" type="GAUGE" xpath="read" />
+</xml-group>
+----
+<1> The data-type is altered in the `xml-object` element from `STRING` to `GAUGE`.
+<2> In this example we associate `aaa` to `10` and `bbb` to `20`.
+<3> A default value can be defined by omitting the `from` attribute in a `xml-mapping` definition.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:194]:
```suggestion
<3> Define a default value by omitting the `from` attribute in a `xml-mapping` definition.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
minor changes. thanks!

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc (hunk 1 above)
Comment (christianpape) [reply docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:6]: Yes. It is.

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc [hunk 8]
```diff
@@ -0,0 +1,37 @@
+= Value Mapping Property Extender
+The Value Mapping property extender provides a mechanism that works like a lookup table for values of a local MIB table column.
+
+== When to use the Value Mapping property extender
+Use the Value Mapping property extender to map an enumerated set of values (for example, strings) to a corresponding set of values.
+This allows OpenNMS to persist changes of string enumeration values as integers over time as time-series.
```
Comment (mmahacek) [inline docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:6]:
```suggestion
The Value Mapping property extender provides a mechanism that works as a lookup table for converting collected values to a different data type.
This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values.
The resulting values are then persisted as time-series data to track state changes over time.
```

**Review by mmahacek — COMMENTED** [review]
Docs section LGTM.

File: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc [hunk 9]
```diff
@@ -0,0 +1,35 @@
+= Value Mapping Property Extender
+The Value Mapping property extender provides a mechanism that works as a lookup table for converting collected values to a different data type.
+This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values.
```
Comment (fooker) [inline docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc:3]:
```suggestion
This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values or the other way around.
```

**Review by fooker — DISMISSED** [review]

**Review by fooker — APPROVED** [review]

**Pull request merged**